### PR TITLE
cake 5: Require php 8.0

### PR DESCRIPTION
**Pull request by othercorey**
This updates the basic configuration to require PHP 8.0.

### CODE REVIEW

**Comment by othercorey** [comment]
@ADmad @markstory @dereuromark @garas not sure if anyone else would like to know.

File: composer.json [hunk 1]
```diff
@@ -22,11 +22,11 @@
         }
     ],
     "require": {
-        "php": ">=7.2.0",
+        "php": ">=8.0",
         "ext-intl": "*",
         "ext-json": "*",
         "ext-mbstring": "*",
-        "cakephp/chronos": "^2.0",
+        "cakephp/chronos": "^2.1.2",
```
Comment (ADmad) [inline composer.json:29]: Should use 2.next for now to avoid errors on the PHP 8.1 build.

Comment (othercorey) [reply composer.json:29]: Are you releasing 2.2. soon?

Comment (ADmad) [reply composer.json:29]: Me personally no. I am no idea what else has gone into Chronos' next minor or what more needs to go. I just fixed the errors on PHP 8.1.

**Comment by markstory** [comment]
Based on our current usage data going to 8.0 would drop a significant number of users. Perhaps that will change by the time we get around to releasing though. Historically ambitious release targets have worked out well.

**Comment by othercorey** [comment]
With 8.0, we get the `mixed` and `static` types as well as compound types `string|int`. IMO, those are the key benefits for moving to 8.0 as they're more compatible with changes made in 8.1+.

There are other future use features like attributes and named arguments. There are some syntax improvements like nullsafe `?->` and `Stringable`.

**Comment by othercorey** [comment]
With no objections from the active members, I'm going to merge this.

**Pull request merged**